contracts/scripts/deploy.js

```diff
@@ -44,89 +33,73 @@ const save = (value, field, subfield = undefined) => {
   fs.writeFileSync(fileName, JSON.stringify(parameters));
 };
 
+// Send 0.01 ETH to oneself and wait for it to be mined, bumping the wallet's nonce
+const doBumpNonceTx = async (wallet) => {
+  const tx = await wallet.sendTransaction({
+    to: wallet.address,
+    value: ethers.utils.parseEther('0.01'),
+  });
+
+  console.log(`Executing nonce bump tx ${tx.hash}`);
+  
+  await tx.wait();
+};
+
 // IIFE async function so "await"s can be performed for each operation
 (async function () {
   try {
     const deployParams = require('./deployParams.json');
     const deployParamsForNetwork = deployParams[network];
 
-    if (!deployParamsForNetwork) {
-      console.log('Invalid network requested', network);
-      save(network, 'actions', 'InvalidNetworkRequested');
-      exit();
-    }
+    // When testing deployment on localhost, the expected deployer is 0x2c399eD6c510C89793C90f7A186FDd2bD1Fd6AdD, which
+    // corresponds to address[0] of the mnemonic "test test test test test test test test test test silent junk".
+    // We could not use the default hardhat mnemonic because the nonce for that address on a forked Rinkeby would
+    // be unpredictable (since lots of people execute txs with that account).
+    const { expectedDeployer, toll, tollCollector, tollReceiver, expectedRegistryNonce } = deployParamsForNetwork;
 
     console.log('Deploying to: ', network);
     save(network, 'actions', 'DeployingContractsToNetwork');
 
-    const [adminWallet] = await ethers.getSigners();
-    save(adminWallet.address, 'admin');
+    const [deployerWallet] = await ethers.getSigners();
+    save(deployerWallet.address, 'deployer');
+
+    if (deployerWallet.address !== expectedDeployer) {
```

Comment:
making sure to compare checksummed addresses since the address in the JSON file could be all lowercase
```suggestion
    if (deployerWallet.address !== ethers.utils.getAddress(expectedDeployer)) {
```

Reply:
Good catch. Done in 59a020255c75f7f0a10324474e7d550e5d30f942